### PR TITLE
User role not nullable

### PR DESCRIPTION
**Pull request by Tirokk**
> [roschaefer](https://github.com/roschaefer) **Authored by [roschaefer](https://github.com/roschaefer)**
_<time datetime="2019-07-08T14:47:06Z" title="Monday, July 8th 2019, 4:47:06 pm +02:00">Jul 8, 2019</time>_
_Merged <time datetime="2019-07-09T18:50:04Z" title="Tuesday, July 9th 2019, 8:50:04 pm +02:00">Jul 9, 2019</time>_
---

## 🍰 Pullrequest
While implementing the signup I discovered this bug. Fortunately, we don't face it until we start to create user accounts.

Includes a refactoring, too.

### Issues

- None

### Todo

- [X] None

### CODE REVIEW

**Comment by Tirokk** [comment]
> [codecov[bot]](https://github.com/apps/codecov) **Authored by [codecov[bot]](https://github.com/apps/codecov)**
_<time datetime="2019-07-08T20:07:17Z" title="Monday, July 8th 2019, 10:07:17 pm +02:00">Jul 8, 2019</time>_

---


# [Codecov](https://codecov.io/gh/Human-Connection/Human-Connection/pull/993?src=pr&el=h1) Report
> :exclamation: No coverage uploaded for pull request base (`master@ae2cf31`). [Click here to learn what that means](https://docs.codecov.io/docs/error-reference#section-missing-base-commit).
> The diff coverage is `n/a`.

File: backend/src/models/User.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  actorId: { type: 'string', allow: [null] },
+  name: { type: 'string', min: 3 },
+  email: { type: 'string', lowercase: true, email: true },
+  slug: 'string',
+  encryptedPassword: 'string',
+  avatar: { type: 'string', allow: [null] },
+  coverImg: { type: 'string', allow: [null] },
+  deleted: { type: 'boolean', default: false },
+  disabled: { type: 'boolean', default: false },
+  role: { type: 'string', default: 'user' },
```
Comment (Tirokk) [inline backend/src/models/User.js:14]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-09T13:42:33Z" title="Tuesday, July 9th 2019, 3:42:33 pm +02:00">Jul 9, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,54 @@
+import uuid from 'uuid/v4'
+
+module.exports = {
+  id: { type: 'string', primary: true, default: uuid }, // TODO: should be type: 'uuid' but simplified for our tests
+  actorId: { type: 'string', allow: [null] },
+  name: { type: 'string', min: 3 },
+  email: { type: 'string', lowercase: true, email: true },
+  slug: 'string',
+  encryptedPassword: 'string',
+  avatar: { type: 'string', allow: [null] },
+  coverImg: { type: 'string', allow: [null] },
+  deleted: { type: 'boolean', default: false },
+  disabled: { type: 'boolean', default: false },
+  role: { type: 'string', default: 'user' },
```


so, basically you changed this to have a default of `user`
the other changes to this `PR` are just refactoring to extract out the models into their own files, correct?

File: backend/src/models/User.spec.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+describe('role', () => {
+  it('defaults to `user`', async () => {
+    const user = await instance.create('User', { name: 'John' })
+    await expect(user.toJson()).resolves.toEqual(
+      expect.objectContaining({
+        role: 'user',
+      }),
+    )
+  })
+})
```
Comment (Tirokk) [inline backend/src/models/User.spec.js:20]: > [mattwr18](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-07-09T13:43:03Z" title="Tuesday, July 9th 2019, 3:43:03 pm +02:00">Jul 9, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -0,0 +1,20 @@
+import Factory from '../seed/factories'
+import { neode } from '../bootstrap/neo4j'
+
+const factory = Factory()
+const instance = neode()
+
+afterEach(async () => {
+  await factory.cleanDatabase()
+})
+
+describe('role', () => {
+  it('defaults to `user`', async () => {
+    const user = await instance.create('User', { name: 'John' })
+    await expect(user.toJson()).resolves.toEqual(
+      expect.objectContaining({
+        role: 'user',
+      }),
+    )
+  })
+})
```


#moresoftwaretests
:smile_cat:

**Review by Mogge — APPROVED** [review]
great job @roschaefer 
I guess that we don't want users with no roles

**Review by Mogge — APPROVED** [review]

**Pull request closed without merging**